.github/workflows/build.yml

```diff
@@ -0,0 +1,29 @@
+name: Continuous Integration
+
+run-name: ${{ github.ref_name }}
+
+on:
+  workflow_dispatch:
+    inputs:
+      ref:
+        type: string
+        description: 'The branch ref to build. Leave empty to use workflow branch.'
+  push:
+    branches:
+      - dev
+      - master
+      - release-*
+      - '[0-9]+.[0-9]+.x'
+    paths-ignore:
+      - '.github/**'
+      - '**/README.md'
+      - '!.github/workflows/build.yml'
+      - '!.github/workflows/_reusable_build.yml'
+
+jobs:
+  call-build-workflow:
+    uses: ./.github/workflows/_reusable_build.yml
+    with:
+      ref: ${{ github.event.inputs.ref || github.ref }}
+      publish:  ${{ (github.event.inputs.ref || github.ref) == 'dev' || (github.event.inputs.ref || github.ref) == 'master' || startsWith((github.event.inputs.ref || github.ref), 'release-') || endsWith((github.event.inputs.ref || github.ref), '.x') }}
```

Comment:
It is worth trying adding publish input in the workflow dispatch without default value and test if the `ithub.event.inputs.publish` is defined to avoid this messy expression.

Reply:
But when will we want to execute this worflow manually? I think the workflow_dispatch is not necessary.

Reply:
I'd say it's best to have the workflow_dispatch in case you want to manually build another branch, or you need to publish a version on a tag out of the branch or anything else.

Anyway, I've come up with a more elegant solution.

Reply:
That's the point of handling base branches builds by github events: we should not trigger those builds manually. If you want to publish a version on a tag that's a dedicated release workflow.

The workflow_dispatch makes the builds more confusing by redefining the inputs used by the reusable build.